src/core/edgeNetwork/sendEdgeNetworkRequestFactory.js

```diff
@@ -12,7 +12,20 @@ governing permissions and limitations under the License.
 
 import { ID_THIRD_PARTY_DOMAIN } from "../../constants/domains";
 import apiVersion from "../../constants/apiVersion";
-import { createCallbackAggregator, noop, uuid } from "../../utils";
+import {
+  createCallbackAggregator,
+  noop,
+  uuid,
+  assign,
+  isNil,
+  isNonEmptyArray
+} from "../../utils";
+
+const notNil = value => !isNil(value) || isNonEmptyArray(value);
+const merge = (acc, value) => {
+  return Array.isArray(value) ? value.reduce(merge, acc) : assign(acc, value);
+};
+const mergeValues = values => values.filter(notNil).reduce(merge, {});
```

Comment:
mergeValues should be defined in a separate file so that we can do easy extensive testing.  It could be a util.

Reply:
ok, I'll extract that in another util